"make test" fails on SELinux enabled systems.

<!--

Welcome! - We kindly ask that you:

  1. Fill out the issue template below - not doing so needs a good reason.
  2. Use the forum if you have a question rather than a bug or feature request.

The forum is at: https://forum.restic.net

NOTE: Not filling out the issue template needs a good reason, as otherwise it
may take a lot longer to find the problem, not to mention it can take up a lot
more time which can otherwise be spent on development. Please also take the
time to help us debug the issue by collecting relevant information, even if
it doesn't seem to be relevant to you. Thanks!

The forum is a better place for questions about restic or general suggestions
and topics, e.g. usage or documentation questions! This issue tracker is mainly
for tracking bugs and feature requests directly relating to the development of
the software itself, rather than the project.

Thanks for understanding, and for contributing to the project!

-->


Output of `restic version`
--------------------------
restic 0.9.6 (v0.9.6-105-g2c9ae595-dirty) compiled with go1.14 on linux/amd64

How did you run restic exactly?
-------------------------------
make test

<!--
This section should include at least:

 * The complete command line and any environment variables you used to
   configure restic's backend access. Make sure to replace sensitive values!

 * The output of the commands, what restic prints gives may give us much
   information to diagnose the problem!
-->

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
none, just running tests


Expected behavior
-----------------
pass

<!--
Describe what you'd like restic to do differently.
-->

Actual behavior
---------------
```
$ make test
go test ./cmd/... ./internal/...
ok  	github.com/restic/restic/cmd/restic	8.297s
--- FAIL: TestMetadataChanged (0.00s)
    testing.go:30: using low-security KDF parameters for test
    config.go:65: disabling check of the chunker polynomial
    archiver_test.go:2052: chdir to /tmp/restic-test-101815698
    archiver_test.go:2080: metadata does not match:
        {*restic.Node}.ExtendedAttributes:
        	-: []restic.ExtendedAttribute(nil)
        	+: []restic.ExtendedAttribute{{Name: "security.selinux", Value: []uint8{0x75, 0x6e, 0x63, 0x6f, 0x6e, 0x66, 0x69, 0x6e, 0x65, 0x64, 0x5f, 0x75, 0x3a, 0x6f, 0x62, 0x6a, 0x65, 0x63, 0x74, 0x5f, 0x72, 0x3a, 0x75, 0x73, 0x65, 0x72, 0x5f, 0x74, 0x6d, 0x70, 0x5f, 0x74, 0x3a, 0x73, 0x30, 0x00}}}
    panic.go:615: chdir back to /home/middelin/go/src/github.com/middelink/restic/internal/archiver
FAIL
FAIL	github.com/restic/restic/internal/archiver	2.249s
```

<!--
In this section, please try to concentrate on observations, so only describe
what you observed directly.
-->

Steps to reproduce the behavior
-------------------------------
Use a Linux system with SELinux active. It will automatically label files when written according to whatever policy is in effect. Hence the archiver sees an unexpected extended attribute "security.selinux"

```
$ lsb_release  -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	CentOS
Description:	CentOS Linux release 8.1.1911 (Core) 
Release:	8.1.1911
Codename:	Core
```
<!--
The more time you spend describing an easy way to reproduce the behavior (if
this is possible), the easier it is for the project developers to fix it!
-->

Do you have any idea what may have caused this?
-----------------------------------------------
See above.


Do you have an idea how to solve the issue?
-------------------------------------------
Ignore the extended attribute if it appears?
Do a preflight check to see if extended attributes get added automatically and if so, expect them?
For the record, disabling SELinux is a complete NO-NO in my book, so that's not a solution.


Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Errr, sure, why not? ^^

<!--
Answering this question is not required, but if you have anything positive to share, please do so here!
Sometimes we get tired of reading bug reports all day and a little positive end note does wonders.
Idea by Joey Hess, https://joeyh.name/blog/entry/two_holiday_stories/
-->

--- a/internal/archiver/archiver_test.go
+++ b/internal/archiver/archiver_test.go
@@ -2071,7 +2071,9 @@ func TestMetadataChanged(t *testing.T) {
 	// set some values so we can then compare the nodes
 	want.Content = node2.Content
 	want.Path = ""
-	want.ExtendedAttributes = nil
+	if len(want.ExtendedAttributes) == 0 {
+		want.ExtendedAttributes = nil
+	}
 
 	want.AccessTime = want.ModTime
 

